### DISCUSSION TITLE
Is it really necessary to restart the library_index.json file?

### DISCUSSION THREAD

**Discussion by rtek1000** [Sloeber/arduino-eclipse-plugin#1509] [Q&A]
Hello,

I noticed that whenever I open the preferences and click apply and close, even though I haven't made any changes, the file is still restarted.

I don't know if it's normal for this library_index.json file to be so big, but I notice that Eclipse has a hard time working with this file on startup, it takes time. It is currently at 27.1MiB.

P.S.: It might be interesting that there is an automatic backup for this file, I already had errors, and I had to delete the file.

[library_index.zip](https://github.com/Sloeber/arduino-eclipse-plugin/files/9867339/library_index.zip)

**Comment by jantje** [comment]
>I noticed that whenever I open the preferences and click apply and close, even though I haven't made any changes, the file is still restarted.

It should not always reload. It should reload is you click apply and close here:
![afbeelding](https://user-images.githubusercontent.com/1175841/198018470-1d039240-4f6f-47bf-ba38-83476cb85329.png)

If you haven't changed anything it is way more safe to select cancel.
What I'm wondering is: **How do you know the json file is reloaded?**

> I don't know if it's normal for this library_index.json file to be so big,

The library_index.json file is provided by Arduino. I highlighted the location from where it is downloaded in the image above. I have no control over the size of the file. You can remove the Arduino provided library json file and you may add your own. 
I have a library json here https://raw.githubusercontent.com/jantje/ArduinoLibraries/master/library_jantje_index.json (also visible on the image above).

> I notice that Eclipse has a hard time working with this file on startup

The "update local json files" check box on the image above toggles the update of the files at Sloeber start-up. In case this is turned off the Sloeber user can not update the files. That is why clicking apply and close there updates the json files (regardless of the flag)
It looks to me as if this option is on in your install.
All json files are downloaded and parsed. This indeed takes some time on my system (As you see I have a big list of json files and they are updated at each and every startup of sloeber :-( ).
What sometimes helps is giving java a bit more memory to play with by increasing the available memory in the .ini file of sloeber/eclipse. Look for something like
```
-Xms256m
-Xmx1024m
```

>It might be interesting that there is an automatic backup for this file, 

Making the downloads of Sloeber more robust has been on Sloeber's todo list for a long time. Unfortunately nobody got to it. It seems there are always more urgent issues.

**Reply by rtek1000** [reply]
> What I'm wondering is: How do you know the json file is reloaded?

![file](https://user-images.githubusercontent.com/8660811/198089871-fd480dc0-f60c-4e4e-9748-64353a1eecc5.png)

I can see the file disappear then the size is incremented. I'm using Linux. It is much faster to compile Arduino than on Windows.


> The "update local json files" check box on the image above toggles the update of the files at Sloeber start-up. In case this is turned off the Sloeber user can not update the files. That is why clicking apply and close there updates the json files (regardless of the flag)
> **It looks to me as if this option is on in your install.**

![pref](https://user-images.githubusercontent.com/8660811/198090756-22863676-b810-4bff-899f-cdca47b4d57b.png)

In fact it is not activated. I had tried to install an [STM32 core](https://github.com/stm32duino/Arduino_Core_STM32), but it didn't complete, that must be why there were these failures. I reinstalled the Sloeber plugin in eclipse and now it seems to be better.

> Making the downloads of Sloeber more robust has been on Sloeber's todo list for a long time. 

About the backup, it might be interesting to at least warn the user to delete the file, if it is malformed.

> What sometimes helps is giving java a bit more memory

I will try to do that.

Thanks.

**Reply by rtek1000** [reply]
> What sometimes helps is giving java a bit more memory to play with by increasing the available memory in the .ini file of sloeber/eclipse. Look for something like

I changed the values from 256 to 8 and from 2048 to 64. There was no change in boot time.

I'm using an Intel I5 (gen 2) processor with DDR3 RAM, and 2Mbps Internet, maybe that's why the Eclipse startup is so hard.

**Comment by jantje** [comment]
>I can see the file disappear then the size is incremented. I'm using Linux. It is much faster to compile Arduino than on Windows.

If I understand correctly you see the file is deleted and then it is downloaded on the file system.

>About the backup, it might be interesting to at least warn the user to delete the file, if it is malformed.

If the file is malformed it should be deleted automagically by Sloeber. I know for sure it works (sometimes) as it has been a pain in the ... to investigate issues.

>I'm using an Intel I5 (gen 2) processor with DDR3 RAM, and 2Mbps Internet, maybe that's why the Eclipse startup is so hard.

Looks not so far of from what I'm using: I'm on a I5 gen 4 low energy processor.
Processor	Intel(R) Core(TM) i5-4250U CPU @ 1.30GHz, 1901 Mhz, 2 Core(s), 4 Logical Processor(s)

**Comment by rtek1000** [comment]
It seems that it is not possible to install Sloeber in the new eclipse IDE:

![eclipse-2209](https://user-images.githubusercontent.com/8660811/198265115-33aacbf7-278b-410f-96e2-b4d9a5f9aa96.png)

In eclipse 2022-03 IDE, it is not possible to add the [STM32 core](https://github.com/stm32duino/Arduino_Core_STM32). But this STM32 core works in Arduino IDE 1.8.19, I tried other cores like Arduino DUE and ESP8266 and it worked.

Any tips to install STM32 core on Sloeber?

**Comment by jantje** [comment]
>Any tips to install STM32 core on Sloeber?

use the product
As far as I know this all works with the product. When mix matching plugin versions one gets all kind of unexpected problems. This is why officially only the product is supported.

**Reply by rtek1000** [reply]
What does Product mean? I tried searching on Google, but I didn't find that reference.

I understand that I shouldn't try to use other platforms in Sloeber, besides the ones that come pre-installed. Is that what the Product means?

I thought that Sloeber should accept the core that the arduino IDE accepts.

**Reply by jantje** [reply]
The product is a all (eclipse/java/cdt/...) containing zip file and has nothing to do with platform in the Arduino way.
You can download from here http://eclipse.baeyens.it/stable.php?OS=Windows
or from here https://github.com/Sloeber/arduino-eclipse-plugin/releases
